### ISSUE TITLE
Crash when UROWID TYPE query

### ISSUE THREAD

**Issue by ad983** [godror/godror#291]
Describe the bug
Hello, I have 3 rows test data in TEST_TABLE. When I query one or two of them, there will be no error. When I query 3 rows of data, the program crashes.

TEST DATA
```
-- Table structure for TEST_TABLE
-- ----------------------------
DROP TABLE "SQLDEV"."TEST_TABLE";
CREATE TABLE "SQLDEV"."TEST_TABLE" (
  "A" NUMBER,
  "S" UROWID(4000)
)
LOGGING
NOCOMPRESS
PCTFREE 10
INITRANS 1
STORAGE (
  INITIAL 65536 
  NEXT 1048576 
  MINEXTENTS 1
  MAXEXTENTS 2147483645
  BUFFER_POOL DEFAULT
)
PARALLEL 1
NOCACHE
DISABLE ROW MOVEMENT
;

-- ----------------------------
-- Records of TEST_TABLE
-- ----------------------------
INSERT INTO "SQLDEV"."TEST_TABLE" ("A", "S") VALUES ('1', '0000001A.0000.0002');
INSERT INTO "SQLDEV"."TEST_TABLE" ("A", "S") VALUES ('2', '0000001A.0000.0002');
INSERT INTO "SQLDEV"."TEST_TABLE" ("A", "S") VALUES ('3', '0000001A.0000.0002');
COMMIT;
COMMIT;

```

go code:

```
package main

import (
	"context"
	"database/sql"
	"fmt"
	"github.com/godror/godror"
	//_ "github.com/godror/godror"
	"time"
)

func main() {
	db, err := connectORACLE("172.19.33.115", "helowin", "sqldev", "xxxx", 1522)
	if err != nil {
		panic(err)
	}

	defer db.Close()

	ctx := context.Background()

	_, err = db.ExecContext(ctx, fmt.Sprintf(`ALTER SESSION SET CURRENT_SCHEMA="%s"`, "SQLDEV"))
	if err != nil {
		fmt.Println("err1:", err.Error())
		return
	}

	sql := `
select 
	t."A",t."S" 
from 
	"TEST_TABLE" t 
`

	rows, err := db.QueryContext(ctx, sql)
	if err != nil {
		fmt.Println("err2:", err.Error())
		return
	}
	defer rows.Close()

	for rows.Next() {
		var id int64
		var text string
		err = rows.Scan(&id, &text)
		if err != nil {
			panic(err.Error())
		}
		fmt.Printf("text: %s\n", text)
	}

}

func connectORACLE(host, serviceName, user, password string, port int) (*sql.DB, error) {

	var P godror.ConnectionParams
	var pp godror.PoolParams
	var cp godror.ConnParams

	pp.SessionTimeout = 30

	P.ConnectString = fmt.Sprintf(`(DESCRIPTION=(connect_timeout=5)(ADDRESS=(PROTOCOL=TCP)(HOST=%v)(PORT=%v))(CONNECT_DATA=(SID=%v)))`, host, port, serviceName)
	P.Username = user
	P.Password = godror.NewPassword(password)
	P.Timezone = time.Local
	P.StandaloneConnection = true
	P.PoolParams = pp
	P.ConnParams = cp
	dd := sql.OpenDB(godror.NewConnector(P))
	dd.SetMaxIdleConns(0)
	dd.SetMaxOpenConns(5)

	return dd, nil
}

```

To Reproduce
At least the offending source code;
the best is a minimal test case that reproduces the error.

Expected behavior

Error output
Copy the error output (panic, stack trace),
or the output of running with DPI_DEBUG_LEVEL=92.

```
GOROOT=/usr/local/go #gosetup
GOPATH=/Users/chengyong/Work/go #gosetup
/usr/local/go/bin/go build -o /Users/chengyong/Library/Caches/JetBrains/GoLand2023.1/tmp/GoLand/___go_build_shuaninfo_com_oracle_test_mytable shuaninfo.com/oracle-test/mytable #gosetup
/Users/chengyong/Library/Caches/JetBrains/GoLand2023.1/tmp/GoLand/___go_build_shuaninfo_com_oracle_test_mytable
text: 0000001A.0000.0002
text: 0000001A.0000.0002
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe5c3d4f]

runtime stack:
runtime.throw({0x4321309?, 0x47085fe?})
        /usr/local/go/src/runtime/panic.go:992 +0x71
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:802 +0x3a9

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x429ad10, 0xc00016f6d0)
        /usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc00016f6a8 sp=0xc00016f670 pc=0x400745c
github.com/godror/godror._Cfunc_dpiRowid_getStringValue(0x4708590, 0xc0000100b0, 0xc0000181d8)
        _cgo_gotypes.go:4692 +0x4c fp=0xc00016f6d0 sp=0xc00016f6a8 pc=0x41ed80c
github.com/godror/godror.(*rows).Next.func4.1(0x0?, 0x1?, 0xc00016f750?)
        /Users/chengyong/Work/go/pkg/mod/github.com/godror/godror@v0.37.0/rows.go:469 +0x7e fp=0xc00016f718 sp=0xc00016f6d0 pc=0x4220f7e
github.com/godror/godror.(*rows).Next.func4()
        /Users/chengyong/Work/go/pkg/mod/github.com/godror/godror@v0.37.0/rows.go:469 +0x25 fp=0xc00016f740 sp=0xc00016f718 pc=0x4220ee5
github.com/godror/godror.(*drv).checkExecNoLOT(0x450e9d8?, 0x1?)
        /Users/chengyong/Work/go/pkg/mod/github.com/godror/godror@v0.37.0/drv.go:254 +0x22 fp=0xc00016f760 sp=0xc00016f740 pc=0x41fddc2
github.com/godror/godror.(*conn).checkExecNoLOT(...)
        /Users/chengyong/Work/go/pkg/mod/github.com/godror/godror@v0.37.0/conn.go:85
github.com/godror/godror.(*rows).Next(0xc000118500, {0xc0000621a0?, 0x2, 0x2})
        /Users/chengyong/Work/go/pkg/mod/github.com/godror/godror@v0.37.0/rows.go:468 +0x319d fp=0xc00016fd30 sp=0xc00016f760 pc=0x42201bd
database/sql.(*Rows).nextLocked(0xc00015e100)
        /usr/local/go/src/database/sql/sql.go:3008 +0x111 fp=0xc00016fd90 sp=0xc00016fd30 pc=0x40e7751
database/sql.(*Rows).Next.func1()
        /usr/local/go/src/database/sql/sql.go:2986 +0x2f fp=0xc00016fdb8 sp=0xc00016fd90 pc=0x40e760f
database/sql.withLock({0x4372438, 0xc00015e130}, 0xc00016fe18)
        /usr/local/go/src/database/sql/sql.go:3437 +0x8c fp=0xc00016fdf8 sp=0xc00016fdb8 pc=0x40e9a8c
database/sql.(*Rows).Next(0xc00015e100)
        /usr/local/go/src/database/sql/sql.go:2985 +0x6f fp=0xc00016fe48 sp=0xc00016fdf8 pc=0x40e758f
main.main()
        /Users/chengyong/Work/go/src/shuaninfo.com/oracle-test/mytable/main.go:42 +0x345 fp=0xc00016ff80 sp=0xc00016fe48 pc=0x4253225
runtime.main()
        /usr/local/go/src/runtime/proc.go:250 +0x212 fp=0xc00016ffe0 sp=0xc00016ff80 pc=0x4039432
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc00016ffe8 sp=0xc00016ffe0 pc=0x4065f41

goroutine 6 [select]:
database/sql.(*DB).connectionOpener(0xc00007f1e0, {0x43728c0, 0xc0000223c0})
        /usr/local/go/src/database/sql/sql.go:1226 +0x8d
created by database/sql.OpenDB
        /usr/local/go/src/database/sql/sql.go:794 +0x18d

Process finished with the exit code 2

```

Your oracle client version
11g

Your godror version
v0.37.0

Your go version
go version go1.19.4 linux/amd64

Your gcc version
gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)

Machine (please complete the following information):
Centos7.6

Additional context

**Comment by cjbj** [comment]
Try setting the environment variable ORA_OCI_NO_OPTIMIZED_FETCH to 1 before starting your program; 
 See https://github.com/oracle/python-cx_Oracle/issues/315#issuecomment-498710438. If this fixes it, then you have hit a known issue, fixed in Oracle DB 19.14

**Comment by ad983** [comment]
@tgulacsi  Thanks ，this is a bug in the Oracle Client libraries.

Setting environment variables can solve this problem

Thanks again

export ORA_OCI_NO_OPTIMIZED_FETCH=1
